src/components/NimbledroidWidget/index.jsx

```diff
@@ -0,0 +1,65 @@
+import { curveLinear } from 'd3';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import MetricsGraphics from 'react-metrics-graphics';
+
+import Widget from '../../quantum/widget';
+
+export default class NimbledroidWidget extends Component {
+  constructor(props) {
+    super(props);
+    this.legendTarget = null;
+  }
+
+  render() {
+    const { profile } = this.props;
+    const labels = Object.keys(profile.data);
+    const data = labels.map(product => profile.data[product]);
+    const target = this.props.targetRatio * profile.lastDataPoints.focus;
+    const status = profile.lastDataPoints.klar / profile.lastDataPoints.focus;
+
+    return (
+      <Widget
+        // This classname is to undo what .criteria-widget:not:first-child sets
+        className={'no-left-margin'}
+        title={profile.title}
+        target='Klar <= Focus + 20%'
+        targetStatus={status < 1 ? 'pass' : 'fail'}
+        status={status < 1 ? 'green' : 'yellow'}
+      >
+        <div
+          style={{
+            display: 'grid',
+            gridTemplateRows: '0.1em 1fr',
+          }}
+        >
+          <div className='legend' ref={ele => this.legendTarget = ele} />
```

Comment:
I suspect this logic is failing, and that's why you're not seeing the legend until you reload the page.

Reply:
I've used [this logic](https://github.com/mozilla/firefox-health-dashboard/blob/acdc71e815ca50b93707d401cfd019b44471b194/src/components/graph-container.js#L86-L87) succesfully in the past. You can see it working in [Android page](https://health.graphics/android).

It's not on reload when the legends will show up. It's on window resize.

If I use `(profile && this.legendTarget)` [1] as in the code mentioned above I won't get the graphs until I resize the window. I left `&& this.legendTarget` so I could at least guarantee the drawing of the graph even if the labels are not drawn.
I need to get a smaller test case.

[1]
```javascript
<div className='legend' ref={ele => this.legendTarget = ele} />
{(profile && this.legendTarget) &&
```

Reply:
Sorry, yeah, I meant resize not reload. I'm not sure, but it could be there is a race condition with this type of logic. Perhaps in the other example, the bug is present but not observable because things happened in a different order.